src/Data/Nat/Properties.agda

```diff
@@ -104,6 +104,28 @@ m ≟ n = map′ (≡ᵇ⇒≡ m n) (≡⇒≡ᵇ m n) (T? (m ≡ᵇ n))
 <⇒<ᵇ (s≤s z≤n)       = tt
 <⇒<ᵇ (s≤s (s≤s m<n)) = <⇒<ᵇ (s≤s m<n)
 
+<ᵇ-reflects-< : ∀ m n → Reflects (m < n) (m <ᵇ n)
+<ᵇ-reflects-< m n with m <ᵇ n | <ᵇ⇒< m n | <⇒<ᵇ {m} {n}
```

Comment:
This and the `≤ᵇ-reflects-≤` lemma below feel like they follow a generic bi-implication pattern that should be extracted to a helper reflects lemma?

Reply:
Good catch. It's literally the proof that if the test is sound & complete for `P`
then it reflects it!